src/slack_interaction_handler.ts

```diff
@@ -319,3 +344,201 @@ export async function handleVolunteerUpdate({
+    "actionTs": payload.action_ts
+  } as SlackModalPrivateMetadata;
+
+  const redisUserInfoKey = `${userId}:${twilioPhoneNumber}`;
+  const userInfo = (await RedisApiUtil.getHash(
+    redisClient,
+    redisUserInfoKey
+  )) as UserInfo;
+
+  if (!userInfo) {
+    modalPrivateMetadata.success = false;
+    modalPrivateMetadata.failureReason = 'no_user_info';
+    await DbApiUtil.logCommandToDb(modalPrivateMetadata);
+    throw new Error(
+      `SLACKINTERACTIONHANDLER.receiveResetDemo: Interaction received for voter who has redisData but not userInfo: active redisData key is ${payload.channel.id}:${payload.message.ts}, userInfo key is ${redisUserInfoKey}.`
+    );
+  }
+
+  let slackView;
+  if (!userInfo.isDemo) {
+    modalPrivateMetadata.success = false;
+    modalPrivateMetadata.failureReason = 'non_demo';
+    await DbApiUtil.logCommandToDb(modalPrivateMetadata);
+    logger.info(
+      `SLACKINTERACTIONHANDLER.receiveResetDemo: Volunteer tried to reset demo on non-demo voter.`
+    );
+    slackView = SlackBlockUtil.getErrorSlackView(
+      'demo_reset_error_not_demo',
+      'This shortcut is strictly for demo conversations only. Please reach out to an admin for assistance.'
+    );
+  } else if (!(payload.channel.id === userInfo.activeChannelId)) {
+    modalPrivateMetadata.success = false;
+    modalPrivateMetadata.failureReason = 'non_active_thread';
+    await DbApiUtil.logCommandToDb(modalPrivateMetadata);
+    logger.info(
+      `SLACKINTERACTIONHANDLER.receiveResetDemo: Volunteer issued reset demo command from #${payload.channel.id} but voter active channel is ${userInfo.activeChannelId}.`
+    );
+    slackView = SlackBlockUtil.getErrorSlackView(
+      'demo_reset_error_not_active_thread',
+      `This voter is no longer active in this thread. Please reach out to the folks at *#${userInfo.activeChannelName}*.`
+    );
+  } else {
+    logger.info(
+      `SLACKINTERACTIONHANDLER.receiveResetDemo: Reset demo command is valid.`
+    );
+
+    // Store the relevant information in the modal so that when the requested action is confirmed
+    // the data needed for the necessary actions is available.
+    slackView = SlackBlockUtil.resetConfirmationSlackView(
+      commandType,
+      modalPrivateMetadata
+    );
+  }
+
+  await SlackApiUtil.renderModal(payload.trigger_id, slackView);
+}
+
+// This function receives the confirmation of the resetting of
+// a voter and does the actual resetting work.
+export async function handleResetDemo(
+  redisClient: PromisifiedRedisClient,
+  modalPrivateMetadata: SlackModalPrivateMetadata
+): Promise<void> {
+  console.log(modalPrivateMetadata);
+  const redisUserInfoKey = `${modalPrivateMetadata.userId}:${modalPrivateMetadata.twilioPhoneNumber}`;
+
+  const slackThreads = (await DbApiUtil.getSlackThreadsForVoter(
+    modalPrivateMetadata.userId,
+    modalPrivateMetadata.twilioPhoneNumber
+  )) as SlackThreadInfo[];
+
+  console.log('slackThreads');
+  console.log(slackThreads);
```

Comment:
Should these console logs be here? There are a handful in this function.

Reply:
Removed. Thanks for catching!